python_modules/dagster/dagster/_core/secrets/loader.py

```diff
@@ -0,0 +1,13 @@
+from abc import ABC, abstractmethod
+from typing import Dict, Optional
+
+from dagster._core.instance import MayHaveInstanceWeakref
+
+
+class SecretsLoader(ABC, MayHaveInstanceWeakref):
+    @abstractmethod
+    def load_env_vars(self, location_name: Optional[str]) -> Dict[str, str]:
+        pass
```

Comment:
nit: `load_env_vars` naming feels a little off to me since this is loading a mapping of values to be injected in to the environment (probably)